### ISSUE TITLE
Neovim float does not get placed relative to the `win` handler passed into `nvim_open_win`

### ISSUE THREAD

**Issue by rmagatti** [neovim/neovim#14667]
- `nvim --version`: `NVIM v0.5.0-dev+a282a177d`
- Operating system/version: MacOS 11.4
- Terminal name/version: Kitty 11.4
- `$TERM`: `xterm-kitty`

~[ ] `vim -u DEFAULTS` (version: ) behaves differently~

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC

:lua vim.api.nvim_open_win(buffer, true, {
    relative='win',
    width=120,
    height=10,
    border="single",
    bufpos={32, 0},
    win=vim.api.nvim_get_current_win()
})

:lua lua vim.api.nvim_open_win(0, true, {relative='win', width=120, height=10, border='single', bufpos={48,0}, win=vim.api.nvim_get_current_win()})
```

### Actual behaviour
Second float window is opened in the wrong position.

### Expected behaviour
Second float opens in a position relative to the `win` handle passed in.
I would have expected the second float to open relative to the first float's position using the provided `bufpos` value in the example below.

Example:
![2021-05-29 02 32 51](https://user-images.githubusercontent.com/2881382/120064195-4c85e400-c028-11eb-9cb2-47f8a9eec5f0.gif)

Edit: so apparently when using a GUI like Goneovim the second float opens in the desired position. So I take it that means this is less likely to be a Neovim issue. I'll keep this open in the meantime in case anyone who sees this ends up having any idea what's happening

Edit2: I'm thinking maybe it actually working is related to the multigrid implementation in Goneovim. I tried other GUIs that also claim multigrid support but no dice.

**Comment by rmagatti** [comment]
Relevant discussion in gitter/matrix
https://matrix.to/#/!JAfPjWAdLCtgeCAwnS:matrix.org/$162267939919884bjIYe:matrix.org?via=matrix.org&via=gitter.im&via=envs.net

TL;DR: this seems to be up to the GUI implementation to handle the float positions correctly. A few GUIs like Goneovim and Uivonim from the ones I've tested do this handling in a way that provides the behaviour I initially expected.

**Comment by folke** [comment]
@rmagatti why was this closed? This problem also occurs in the terminal. I've opened an issue about it here https://github.com/neovim/neovim/issues/14735

**Comment by rmagatti** [comment]
@folke my understanding from the Matrix chat was that it was not up to Neovim to fix, which I also found to be odd but it's what led me to close this anyway.

Thanks for opening another issue about it, hope it gets more traction than this one.

**Comment by bfredl** [comment]
The TUI case needs to be handled by nvim itself, indeed.
